File: app/src/main/java/br/pedroso/upcomingmovies/di/ApplicationComponent.java
package br.pedroso.upcomingmovies.di;

import javax.inject.Singleton;

import br.pedroso.upcomingmovies.network.di.NetworkModule;
import br.pedroso.upcomingmovies.repository.di.RepositoryModule;
import br.pedroso.upcomingmovies.domain.MoviesRepository;
import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class, NetworkModule.class, RepositoryModule.class})
public interface ApplicationComponent {
    MoviesRepository moviesRepository();
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/di/ApplicationComponent.kt
package br.pedroso.upcomingmovies.di

import br.pedroso.upcomingmovies.domain.MoviesRepository
import br.pedroso.upcomingmovies.network.di.NetworkModule
import br.pedroso.upcomingmovies.repository.di.RepositoryModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class, NetworkModule::class, RepositoryModule::class])
interface ApplicationComponent {
    fun moviesRepository(): MoviesRepository
}
